ironfish-rust/src/keys/mod.rs

```diff
@@ -197,12 +185,22 @@ impl SaplingKey {
     /// a seed. This isn't strictly necessary for private key, but view keys
     /// will need a direct mapping. The private key could still be generated
     /// using bip-32 and bip-39 if desired.
-    pub fn words_spending_key(&self, language: Language) -> Result<String, IronfishError> {
+    pub fn to_words(&self, language: Language) -> Result<String, IronfishError> {
         let mnemonic = Mnemonic::from_entropy(&self.spending_key, language)
             .map_err(|_| IronfishError::InvalidEntropy)?;
         Ok(mnemonic.phrase().to_string())
     }
 
+    /// takes words and turns them into spending key
+    pub fn from_words(words: String, language: Language) -> Result<Self, IronfishError> {
+        let mnemonic = Mnemonic::from_phrase(&words, language)
+            .map_err(|_| IronfishError::InvalidPaymentAddress)?;
```

Comment:
This would be easier to use if we implemented the From error type wrapper on IronfishError the way we do for other error types, and just wrap the error. Given this already existed, just pointing it out, not asking for it to be changed. Then you're able to skip the `map_err`, since `?` implicitly calls `.into()` if i remember correctly

Reply:
Added it as a follow up (along with returning Mnemonic rather than string in rust layer)